tools/toolchains/__init__.py

```diff
@@ -512,6 +512,10 @@ def is_ignored(self, file_path):
                 return True
         return False
 
+    def add_ignore_patterns(self, root, base_path, patterns):
```

Comment:
Would you mind adding a doc string to this? `root` and `base_path` are a bit confusing.